### ISSUE TITLE
[Feature request] Full support for proxy parameter

### ISSUE THREAD

**Issue by burbilog** [Genymobile/gnirehtet#533]
I tried to compile this https://github.com/Genymobile/gnirehtet/pull/504 and it works, at least for me it's usable this way.

But  I had to compile in my own proxy server ip. Proxy server setting should be passed from server command line with parameter like -P http://ip:3128 from server side. 

Please, implement it, it's a great feature.

**Comment by VaradGupta23** [comment]
You're absolutely right — adding full support for a --proxy or -P parameter in gnirehtet would greatly improve flexibility, especially in corporate, university, or firewalled networks where:

Direct access to the internet is not allowed,

Only proxy (HTTP/SOCKS) access is permitted,

Or users want to route traffic through custom anonymizing proxies (like Tor, Privoxy, etc.)

Current Limitation
Right now, if you want gnirehtet traffic (i.e., Android device's reverse-tethered traffic) to go through a proxy, your only options are:

Manually configure your PC’s proxy

Set up a transparent proxy + iptables/redsocks

Recompile gnirehtet with hardcoded proxy logic (like you did in [#504](https://github.com/Genymobile/gnirehtet/issues/504))

There’s no runtime flag like:

gnirehtet relay -P http://127.0.0.1:3128
…to pass a proxy setting.

 Proposed Feature
CLI Flag
Allow:


gnirehtet relay --proxy http://<proxy-ip>:<port>
Or:

gnirehtet run --proxy socks5://127.0.0.1:9050
Where:

Supported schemes: http://, https://, socks5://

Also optional basic auth: http://user:pass@proxy.example.com:3128

 How It Could Work Internally
The relay could wrap the outgoing socket connections using Java’s Proxy class or manually manage it per stream.

Proxy logic would be injected in the host-side relay, not the Android APK (so no APK change needed).

Optionally: allow PAC (proxy auto-config) or NO_PROXY environments.
